### PR TITLE
adding unique ids to other content types

### LINKED ISSUE #104: Add unique identifier for different content types
Currently we only set post_id, but now that we are going to allow prompts on non post pages, for the metrics, we should be able to identify different content types the prompt can exist on.

Thinking posts would stay the same ie `123`, but for pages, category pages, and such, this is the convention we'll follow:

Posts: `post=123`
Pages: `page=123`
Custom content types: `custom_content_type=12`
Home: `home`

Archives:

Categories: `category=5`
Post archive: `archive`
Author: `author=12`
Tag: `tag=32`
Term: `term=155`

### PR DESCRIPTION
**Pull request by rachelslurs**
### Description
This changes the convention of how we store where the login prompt occurs, since we're no longer limited to posts alone. These will be set according to these rules:
Posts: `123` <- putting it back to what it was for the `/verify` endpoint to not get upset.
Pages: `page=123`
Custom content types: `custom_content_type=12`
Home: `home`

**Archives:**
Categories: `category=5`
Post archive: `archive`
Author: `author=12`
Tag: `tag=32`
Term: `term=155`

:white_check_mark: Fixes #104 

### Instructions for review
1. Navigate to the home page and in the console.log, evaluate `Pico.content_id` and see that it is 'home'
2. Navigate to an archive page, and see that `Pico.content_id` resolves to 'archive'
3. Navigate to a category archive page, and see that `Pico.content_id` resolves to 'category={id}'
4. Navigate to an author archive page, and see that `Pico.content_id` resolves to 'author={id}'
5. Navigate to a page, and see that `Pico.content_id` resolves to 'page={id}'
6. Navigate to a post, and see that `Pico.article_id` resolves to '{id}'
7. Navigate to a URL that doesn't exist ie `/asfas`, and see that `Pico.content_id` resolves to the query string `pagename=asfas`

:pray: to be blessed and merged in by @hoppy-kamper

### CODE REVIEW

File: includes/class.widget.php [hunk 1]
```diff
@@ -128,50 +128,89 @@ public static function load_widget_assets() {
...
         wp_enqueue_script( 'read-more.js');
-	}
+    }
+
+	/**
+	 * Setting the unique identifier for the current content type being requested
+     * Loosely based on source from https://core.trac.wordpress.org/browser/tags/4.9/src/wp-includes/post-template.php#L589
+	 * @return [string]
+	 */
+	public static function get_unique_content_identifier() {
```
Comment (hoppy-kamper) [inline includes/class.widget.php]: For `/nonsense` endpoints (that return 404 from wordpress) pico.article_id is returing an empty string `""` for me, which I'm fine with but I think it's different than the PR description.

screen shot 2018-05-15 at 7 01 11 pm

Comment (hoppy-kamper) [inline includes/class.widget.php]: I also made a page and went to it, but the `Pico.article_id` is still an empty string and no Articles record is being saved to the db

tbh I don't know much about Wordpress so I might be doing things wrong, but let me know if I can help in any way.

screen shot 2018-05-15 at 7 04 04 pm

Comment (rachelslurs) [reply includes/class.widget.php]: Sorry! Updated the variable to `Pico.content_id` for non articles

**Comment by UnbrandedTech** [comment]
Other then having a unique id, one of the things that is nice about the `post_id` is that we could redirect them to `http://wordpress.local/?p={article.ref_id}`, it would resolve to a page. 

With the user overview page and some analytics down the road, about what page they converted on, etc, is it possible to have a custom endpoint that we can request the url based on this content id?

**Comment by rachelslurs** [comment]
I ended up leaving post id as is (ie `123`, and for other content types, I have them set to a value we can query by - ie page=12, category=2

**Pull request merged**